README.md

```diff
@@ -1 +1,55 @@
-# user_auth
\ No newline at end of file
+# Setup
+If you do not have the docker postgres image, it can be pulled from the docker container repo.
+```
+$ docker pull postgres
+```
+
+Run a docker image with postgres. You can run without the `-v` arg and value to prevent data from persisting and not need to create the directories.
+
+```
+$ docker run --rm --name pg-docker -e POSTGRES_PASSWORD=docker -d -p 5432:5432 -v $HOME/docker/volumes/userAuth:/var/lib/postgresql/data postgres
```

Comment:
This is personal preference for me and I mention it only in case it's useful to you.

I prefer using `docker-compose`, even if it's only for one service. It kinda abstracts away everything for me.

```yaml
-- docker-compose.yml
-- You can also put volumes here, but I'm not a volume guy
version: '3.1'
services:
  postgres:
    image: postgres:11.2
    environment:
      - POSTGRES_USER=$USER
    ports:
      - 5432:5432
```

Notice I didn't put a password and I used the current user ('david' for me, 'lee' for you) as the database user. This is so you can do:
```sh
postgresql://localhost/user_auth
# instead of
postgresql://postgres:docker@localhost:5432/user_auth
```

For updating the db you can just:
```sh
createdb -h localhost user_auth
psql -h localhost -f schema.sql user_auth
```

Because you're already publishing those ports, no need to go through docker and its cryptic commands

Reply:
This is awesome! I changed all of it but the one thing that is giving me trouble is my dump command. I originally had
```
docker exec pg-docker pg_dump -s -U postgres user_auth > schema.sql
```
but this uses `-U postgres` which doesn't exist anymore. If I remove it and run
```
docker exec user_auth_postgres_1 pg_dump -s user_auth > schema.sql
```
It tells me `role "root" does not exist`. If I try with the current user like
```
docker exec user_auth_postgres_1 pg_dump -s -U $USER user_auth > schema.sql
```
this works, but then my dump file that anyone else would restore from alters the tables to owner leecostello. I feel like that is incorrect. Am I missing a step and the `postgres` user should still exist or is there something else I should be doing?

note: the switch from `pg-docker` to `user_auth_postgres_1` is because the name of the container changed

In psql `\du` only shows one row, with `Role name` = `leecostello`

Reply:
I read this _after_ making comments in the review, lol. If you execute `pg_dump` through docker then your user is `root` (because you're in docker). I gave you a command on the review that I think should work. Let me know if it doesn't work or if you have trouble understanding something.

Reply:
This worked much better! Only change I had to make was the version of the docker image. It had mismatched versions of the database and pg_dump.